src/EFCore.PG.NodaTime/Storage/Internal/NpgsqlNodaTimeTypeMappingSourcePlugin.cs

```diff
@@ -0,0 +1,199 @@
+{
+    public class NpgsqlNodaTimeTypeMappingSourcePlugin : IRelationalTypeMappingSourcePlugin
+    {
+        public ConcurrentDictionary<string, RelationalTypeMapping[]> StoreTypeMappings { get; }
+        public ConcurrentDictionary<Type, RelationalTypeMapping> ClrTypeMappings { get; }
+
+        #region TypeMapping
+
+        [NotNull] readonly TimestampInstantMapping _timestampInstant = new TimestampInstantMapping();
+        [NotNull] readonly TimestampLocalDateTimeMapping _timestampLocalDateTime = new TimestampLocalDateTimeMapping();
+
+        [NotNull] readonly TimestampTzInstantMapping _timestamptzInstant = new TimestampTzInstantMapping();
+        [NotNull] readonly TimestampTzZonedDateTimeMapping _timestamptzZonedDateTime = new TimestampTzZonedDateTimeMapping();
+        [NotNull] readonly TimestampTzOffsetDateTimeMapping _timestamptzOffsetDateTime = new TimestampTzOffsetDateTimeMapping();
+
+        [NotNull] readonly DateMapping _date = new DateMapping();
+        [NotNull] readonly TimeMapping _time = new TimeMapping();
+        [NotNull] readonly TimeTzMapping _timetz = new TimeTzMapping();
+        [NotNull] readonly IntervalMapping _period = new IntervalMapping();
+
+        [NotNull] readonly NpgsqlRangeTypeMapping _timestampLocalDateTimeRange;
+        [NotNull] readonly NpgsqlRangeTypeMapping _timestampInstantRange;
+        [NotNull] readonly NpgsqlRangeTypeMapping _timestamptzInstantRange;
+        [NotNull] readonly NpgsqlRangeTypeMapping _timestamptzZonedDateTimeRange;
+        [NotNull] readonly NpgsqlRangeTypeMapping _timestamptzOffsetDateTimeRange;
+        [NotNull] readonly NpgsqlRangeTypeMapping _dateRange;
+
+        #endregion
+
+        /// <summary>
+        /// Constructs an instance of the <see cref="NpgsqlNodaTimeTypeMappingSourcePlugin"/> class.
+        /// </summary>
+        public NpgsqlNodaTimeTypeMappingSourcePlugin()
+        {
+            _timestampLocalDateTimeRange = new NpgsqlRangeTypeMapping("tsrange", typeof(NpgsqlRange<LocalDateTime>), _timestampLocalDateTime);
+            _timestampInstantRange = new NpgsqlRangeTypeMapping("tsrange", typeof(NpgsqlRange<Instant>), _timestampInstant);
+            _timestamptzInstantRange = new NpgsqlRangeTypeMapping("tstzrange", typeof(NpgsqlRange<Instant>), _timestamptzInstant);
+            _timestamptzZonedDateTimeRange = new NpgsqlRangeTypeMapping("tstzrange", typeof(NpgsqlRange<ZonedDateTime>), _timestamptzZonedDateTime);
+            _timestamptzOffsetDateTimeRange = new NpgsqlRangeTypeMapping("tstzrange", typeof(NpgsqlRange<OffsetDateTime>), _timestamptzOffsetDateTime);
+            _dateRange = new NpgsqlRangeTypeMapping("daterange", typeof(NpgsqlRange<LocalDate>), _date);
+
+            var storeTypeMappings = new Dictionary<string, RelationalTypeMapping[]>(StringComparer.OrdinalIgnoreCase)
+            {
+                { "timestamp", new RelationalTypeMapping[] { _timestampInstant, _timestampLocalDateTime } },
```

Comment:
Just a thought, but since the key is just a string here, it might look a bit more terse to use `["key"] = ...` style assignment. 

That said, I think this is consistent with other mappers, so either way is fine by me.

Reply:
I think we discussed this once, looks pretty much identical to me... Am going to leave as-is for now but let me know if you want to revisit.

Reply:
Nope, that's fine by me! I'd written out a lengthy dictionary field at work last week, so the syntax was just fresh on my mind.